lib/vtk/commands/socks/setup.rb

```diff
@@ -0,0 +1,297 @@
+          FileUtils.chmod 0o600, "#{ssh_dir}/config"
+        end
+
+        def backup_existing_ssh_config
+          return true unless File.exist? ssh_config_path
+
+          if File.exist? "#{ssh_config_path}.bak"
+            log "!!! ERROR: Could not make backup of #{pretty_ssh_config_path} as #{pretty_ssh_config_path}.bak " \
+              'exists. Aborting.'
+            exit 1
+          end
+
+          FileUtils.mv ssh_config_path, "#{ssh_config_path}.bak"
+        end
+
+        def configure_ssh_config_with_keychain
+          return unless macos?
+          return if ssh_config_configured_with_keychain?
+          return true unless prompt.yes? "----> #{pretty_ssh_config_path} missing Keychain configuration. Add now?"
+
+          keychain_config = <<~CFG
+
+            # Maintain SSH keys in macOS Keychain
+            Host *
+              UseKeychain yes
+              AddKeysToAgent yes
+              IdentityFile #{pretty_ssh_key_path}
+          CFG
+
+          IO.write ssh_config_path, keychain_config, mode: 'a'
+        end
+
+        def download_ssh_config
+          log 'Downloading and checking ssh/config...'
+
+          repo_url = 'https://github.com/department-of-veterans-affairs/devops.git'
+          `git clone --quiet#{' --depth 1' if macos?} --no-checkout --filter=blob:none #{repo_url} /tmp/dova-devops`
+          `cd /tmp/dova-devops; git checkout master -- ssh/config`
+        end
+
+        def ssh_config_clean_up
+          FileUtils.rm_rf '/tmp/dova-devops'
+        end
+
+        def ssh_agent_add
+          if macos?
+            `ssh-add -K 2> /dev/null; ssh-add -K #{ssh_key_path} 2> /dev/null`
+          elsif ubuntu?
+            `[ -z "$SSH_AUTH_SOCK" ] && eval "$(ssh-agent -s)";
+              ssh-add 2> /dev/null; ssh-add #{ssh_key_path} 2> /dev/null`
+          end
+        end
+
+        def test_ssh_connection
+          output.print '----> Testing SOCKS SSH connection...'
+
+          ssh_output = `ssh -F #{ssh_config_path} -o ConnectTimeout=5 -q socks -D #{port} exit`.chomp
+          if ssh_output == 'This account is currently not available.'
+            output.puts ' ✅'
+          else
+            output.puts ' ❌'
+            log '  ERROR: SSH Connection to SOCKS server unsuccessful. Error message:'
+            output.puts `ssh -F #{ssh_config_path} -o ConnectTimeout=5 -vvv socks -D #{port} -N`
+            exit 1
+          end
+        end
+
+        def configure_system_boot
+          return false unless macos?
+          return true unless
+          `launchctl list | grep #{launch_agent_label}`.empty?
+          return false unless prompt.yes? '----> Start SOCKS on system boot?'
+
+          install_autossh
+          install_launch_agent
+
+          true
+        end
+
+        def launch_agent_label
+          @launch_agent_label ||= begin
+            launch_agent_label = 'gov.va.socks'
+            launch_agent_label += "-test-#{rand 1000}" if ENV['TEST'] == 'test'
```

Comment:
This is a bit of a hack to get the tests to work right without stepping on the toes of the dev's existing LaunchAgent. I was tired of passing args through the command. Having `--launch-agent-label` flag seemed really dumb. Another solution here would be to use `ENV['LAUNCH_AGENT_LABEL']`. That might be a better option. /shrug